### DISCUSSION TITLE
[Idea] Winget-AutoUpdate-ModRepo [Edit: Repo closed]

### DISCUSSION THREAD

**Discussion by Romanitho** [Romanitho/Winget-AutoUpdate#272] [MODs]
We could create a new project in GitHub based on the same mechanism of https://github.com/microsoft/winget-pkgs/tree/master/manifests where we could share and approve community mods with basics (or necessary) actions. WAU could be linked to that repo (by default or with WAU argument on install [or GPO 😇]) and could download the MODS "on demand" (to avoid to get all the available ps1 scripts of all apps) when an application update is available. The MODs discussion could be transferred to this new repo.
@KnifMelti MODs is your idea, so...

**Comment by Romanitho** [comment]
a first approach:  
https://github.com/Romanitho/Winget-AutoUpdate-ModsRepo/tree/draft

**Reply by KnifMelti** [reply]
Interesting idea!
Linking **WAU** to it by default is in my mind not an option.
Modifications are critical components and a lot could go wrong if not accurate implemented and it changes all the time depending on the manifests!
The upside is the community management if enough people engage in the **ModsRepo**!
Less monitoring for each one, different ideas to implement a mod, less to manage, etc...

**Reply by KnifMelti** [reply]
From a security standpoint:
In a domain with now 5000+ WAU-managed clients, we can't allow a PS script to do anything in SYSTEM context that we don't have the control of ourself.
Who's gonna moderate the Repo?`
On a bad day that individual could do anything in the scripts, download trojans etc...
...and all the linked computers would be compromised the next time a winget app got upgraded and modified.

**Reply by Romanitho** [reply]
Totally agree. That's mainly why i discourage WAU to be set with self update on corporate environment. And mods repo should not be used by enterprises.
Maybe we could use this repo only for sharing 🤷🏼‍♂️

**Reply by KnifMelti** [reply]
Well, as long as it's an install argument everyone can make the decision themself.
Who's gonna moderate the Repo? 😉

**Reply by KnifMelti** [reply]
Is it moderated by you in every commit (tested etc...)

**Reply by Romanitho** [reply]
exactly, everyone make the decision themself. But today this is already the case with WAU. Everyone activating WAU Self-Update means they trust our work.  
Regarding an eventual WAU Mod repo, I agree it will be a hard work to approve all the scripts 😅
But, it could be a template repo that people fork for their need and filter the scripts they want to use. Anyway, we will not have time to maintain that for sure ^^ So, probably not that good idea in fact 😂

**Reply by Romanitho** [reply]
[Repo closed]

**Comment by KnifMelti** [comment]
ChatGPT said:
`Invoke-Expression (Invoke-WebRequest -Uri https://raw.githubusercontent.com/Romanitho/Winget-AutoUpdate-ModsRepo/draft/MODs/S/ShareX/ShareX/ShareX.ShareX-preinstall.ps1 -UseBasicParsing).Content`

**Reply by Romanitho** [reply]
I think you're right, maybe the Mod repo could be an idea to centralise community scripts, but not be used as an automatic system.

By the way, we could also secure the mods folder in WAU to avoid users to add own scripts.

**Reply by KnifMelti** [reply]
Ah, didn't notise that rights wasn't locked to admins.. ..that should be done!

**Reply by KnifMelti** [reply]
#275
